File: Java/AddTwoNumbers.java
class ListNode {
    int val;
    ListNode next;
    ListNode(int x) { val = x; }
}

public class AddTwoNumbers {
    public ListNode addTwoNumbers(ListNode l1, ListNode l2) {
        int add = 0;
        ListNode head = new ListNode(0);
        ListNode curr = head;
        while (l1 != null || l2 != null) {
            int res = add;
            if (l1 != null) {
                res += l1.val;
                l1 = l1.next;
            }
            if (l2 != null) {
                res += l2.val;
                l2 = l2.next;
            }
            ListNode node = new ListNode(res % 10);
            curr.next = node;
            curr = curr.next;
            add = res / 10;
        }
        if (add == 1) {
            curr.next = new ListNode(1);
        }
        return head.next;
    }
}

File: Python/1-50/AddTwoNumbers.py
# Definition for singly-linked list.
# class ListNode(object):
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution(object):
    def addTwoNumbers(self, l1, l2):
        """
        :type l1: ListNode
        :type l2: ListNode
        :rtype: ListNode
        """
        vhead = ListNode(0)
        p = vhead
        carry = 0
        while l1 or l2 or carry > 0:
            v = carry
            if l1:
                v += l1.val
                l1 = l1.next
            if l2:
                v += l2.val
                l2 = l2.next
            carry = v / 10;
            p.next = ListNode(v % 10)
            p = p.next
        return vhead.next
